python/lsst/cp/pipe/defects.py

```diff
@@ -529,6 +546,141 @@ def findHotAndColdPixels(self, exp, imageType, setMask=False):
             footprintList += mergedSet.getFootprints()
 
         defects = measAlg.Defects.fromFootprintList(footprintList)
+
+        defects = self.maskBlocksIfIntermitentBadPixelsInColumn(defects)
+
+        return defects
+
+    def maskBlocksIfIntermitentBadPixelsInColumn(self, defects):
+        """Mask blocks in a column if there are on-and-off bad pixels
+
+        If there's a column with on and off bad pixels, mask all the pixels in between,
+        except if there is a large enough gap of consecutive good pixels between two
+        bad pixels in the column.
+
+        Parameters
+        ---------
+        defects: `lsst.meas.algorithms.Defect`
+            The defcts found in the image so far
+
+        Returns
+        ------
+        defects: `lsst.meas.algorithms.Defect`
+            If the number of bad pixels in a column is not larger or equal than
+            self.config.badPixelColumnThreshold, the iput list is returned. Otherwise,
+            the defects list returned will include boxes that mask blocks of on-and-of
+            pixels.
+        """
+        # Get the x, y values of the defects.
+        x, y = [], []
+        extensionX = []
+        for defect in defects:
+            bbox = defect.getBBox()
+            x.append(bbox.getMinX())
+            y.append(bbox.getMinY())
+            dimensions = bbox.getDimensions()
+            extensionX.append(dimensions[0])
+        x = np.array(x)
+        y = np.array(y)
+        extensionX = np.array(extensionX)
+        # Find the defects with same "x" (vertical) coordinate (column).
+        unique, counts = np.unique(x, return_counts=True)
```

Comment:
Maybe I'm misreading the code but I'm still a bit worried that the code cares about the edge of the defect region and that if some defects were bigger than single pixels and started a few pixels over that they would be missed. If the bad pixels were converted into that mask and that mask was collapsed into a 1-D histogram, would that be enough to tell you if a majority of pixels in the column were masked out so you could mask the whole column? You could then fix up the mask itself and then generate a new set of defects from the mask.

Reply:
I added a few lines that loop over each defect bounding box to obtain the coordinates (x,y) of each individual bad pixel in the amp.

Reply:
This still all looks really complicated. Did you discount the approach of generating the mask and then scanning quickly through each column looking for gaps?  Do you know you handle overlapping bounding boxes properly? The tests don't look to me like they deal with cases other than your hand crafted situation.

Reply:
Thank you for your comment. I adopted this approach because it seemed simpler to me than generating the mask. I am assuming that going through all the boxes and then going through each box to get the (x,y) coordinates of all the bad pixels would make sure that no bad pixel was missed, regardless whether their original bounding boxes overlapped or not. Perhaps I am not understanding your concern about overlapping bounding boxes. Would you mind elaborating on that, please?  Regarding the test, I designed it in order to address the charge of the ticket (blocking on-and-off bad pixels in a column) and the comments above related to possible multiple columns (which was a concern when I was using the origin coordinate of the boxes---x_min and y_min---but that is addressed when using single pixel coordinates). Do you have another case (or other cases) that perhaps I could test?